File: algorithms/array/containsduplicate217/solution.js
/*! *****************************************************************************
@author Heming
founded at 2020-12-01 21:43:39
created by IntelliJ IDEA
description: LeetCode 217：存在重复元素
from：https://leetcode-cn.com/problems/contains-duplicate/
***************************************************************************** */

/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
  const set = new Set(nums);
  return set.size === nums.length;
};

File: algorithms/array/containsduplicate217/Solution.java
package blogandquestion.algorithms.array.containsduplicate217;

import java.util.HashSet;
import java.util.Set;

/**
 * @author Heming
 * created by IntelliJ IDEA
 * founded at 2020/12/01 21:43:25
 * description:
 */
public class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int i : nums) {
            set.add(i);
        }
        return set.size() != nums.length;
    }
}
